inference-engine/tests/functional/shared_test_classes/src/single_layer/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -37,25 +40,51 @@ std::string CTCGreedyDecoderSeqLenLayerTest::getTestCaseName(
 }
 
 void CTCGreedyDecoderSeqLenLayerTest::SetUp() {
-    InferenceEngine::SizeVector inputShapes;
+    InferenceEngine::SizeVector inputShape;
+    int sequenceLengths;
     InferenceEngine::Precision dataPrecision, indicesPrecision;
     int blankIndex;
     bool mergeRepeated;
-    std::tie(inputShapes,
-        dataPrecision,
-        indicesPrecision,
-        blankIndex,
-        mergeRepeated,
-        targetDevice) = GetParam();
+    std::tie(inputShape,
+             sequenceLengths,
+             dataPrecision,
+             indicesPrecision,
+             blankIndex,
+             mergeRepeated,
+             targetDevice) = GetParam();
 
     auto ngDataPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(dataPrecision);
     auto ngIdxPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(indicesPrecision);
-    auto paramsIn = ngraph::builder::makeParams(ngDataPrc, { inputShapes });
+    auto paramsIn = ngraph::builder::makeParams(ngDataPrc, { inputShape });
     auto paramOuts = ngraph::helpers::convert2OutputVector(
         ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(paramsIn));
 
+    const auto sequenceLenNode = [&] {
```

Comment:
Looks like, we can move this code to the builder

Reply:
No, this code generates data for the second input, and it has random values. Moving this part to the builder function will give it unpredictable behavior. With the same inputs, a different ngraph::function will be generated each time. By that I mean that the reference output will be different.